### PR TITLE
exp/emove subrouter

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description

Remove usage of subrouter, add routes with listenpath prefixed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5480/merge
Commit: c382084149f421ae71c16510fd06440cef8caf9e 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6020883748)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5480/merge
Commit: 2dc6bb33916b24642f2f9ba995958a6026d0bd2f 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6021071688)

File: gateway/server.go [hunk 1]
```diff
@@ -912,6 +941,26 @@ func handleCORS(router *mux.Router, spec *APISpec) {
 	}
 }
 
+func getCORSHandler(spec *APISpec) func(h http.Handler) http.Handler {
+
+	if spec.CORS.Enable {
```
Comment (titpetric) [inline gateway/server.go:946]: reverse the condition and early exit on spec == nil || !spec.CORS.Enable, decrease indentation

File: gateway/server.go [hunk 2]
```diff
@@ -694,11 +720,14 @@ func generateOAuthPrefix(apiID string) string {
 // Create API-specific OAuth handlers and respective auth servers
 func (gw *Gateway) addOAuthHandlers(spec *APISpec, muxer *mux.Router) *OAuthManager {
 
-	apiAuthorizePath := "/tyk/oauth/authorize-client{_:/?}"
-	clientAuthPath := "/oauth/authorize{_:/?}"
-	clientAccessPath := "/oauth/token{_:/?}"
-	revokeToken := "/oauth/revoke"
-	revokeAllTokens := "/oauth/revoke_all"
+	prefixedWithListenPath := func(endpoint string) string {
```
Comment (titpetric) [inline gateway/server.go:723]: may as well just name it `p` or some short var - does the long function name add to readability here?

File: gateway/server.go [hunk 3]
```diff
@@ -421,7 +424,9 @@ func (gw *Gateway) setupGlobals() {
 		NewRelicApplication = gw.SetupNewRelic()
 	}
 
-	gw.readGraphqlPlaygroundTemplate()
+	if playgroundTemplate == nil {
```
Comment (titpetric) [inline gateway/server.go:427]: I'd add an additional condition of spec.GraphQL.GraphQLPlayground.Enabled, we already know it should be behind that, or move the block to graphql scope, use a sync.Once...

File: gateway/reverse_proxy.go [hunk 4]
```diff
@@ -340,7 +340,7 @@ func (gw *Gateway) TykNewSingleHostReverseProxy(target *url.URL, spec *APISpec,
 		},
 		sp: sync.Pool{
 			New: func() interface{} {
-				buffer := make([]byte, 32*1024)
+				buffer := make([]byte, 2*1024)
```
Comment (titpetric) [inline gateway/reverse_proxy.go:343]: Add a config/env var to configure this buffer size to a reasonable amount? Small api responses require smaller buffers, in different scenarios a larger buffer would make sense. We're sure it doesn't leak, gets improved in latest mux/fork, but it does create allocation pressure so tuning this value is important

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5480/merge
Commit: 3a71e5be9c20ad91bf1abc46b96b822ab5dfc3fb 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6021763488)

File: gateway/server.go [hunk 5]
```diff
@@ -665,6 +642,55 @@ func (gw *Gateway) loadControlAPIEndpoints(muxer *mux.Router) {
...
+
+// loadControlAPIEndpoints loads the endpoints used for controlling the Gateway.
+func (gw *Gateway) loadControlAPIEndpoints(muxer *mux.Router) {
+	hostname := gw.GetConfig().HostName
+	if gw.GetConfig().ControlAPIHostname != "" {
+		hostname = gw.GetConfig().ControlAPIHostname
+	}
+
+	if muxer == nil {
+		cp := gw.GetConfig().ControlAPIPort
```
Comment (titpetric) [inline gateway/server.go:657]: I don't like the indentation here, it doesn't look like a good API with several fallbacks. A smaller function for this block would clear up the logic.

Something like:
`func (gw *Gateway) getProxyMux() (*mux.Router, error)`

**Pull request closed without merging**